builtin/providers/google/resource_compute_target_https_proxy_test.go

```diff
@@ -7,46 +7,73 @@ import (
 	"github.com/hashicorp/terraform/helper/acctest"
 	"github.com/hashicorp/terraform/helper/resource"
 	"github.com/hashicorp/terraform/terraform"
+	"google.golang.org/api/compute/v1"
 )
 
 func TestAccComputeTargetHttpsProxy_basic(t *testing.T) {
+	var targetHttpsProxy compute.TargetHttpsProxy
+	var resourceName = fmt.Sprintf("httpsproxy-test-%s", acctest.RandString(10))
 
 	resource.Test(t, resource.TestCase{
 		PreCheck:     func() { testAccPreCheck(t) },
 		Providers:    testAccProviders,
 		CheckDestroy: testAccCheckComputeTargetHttpsProxyDestroy,
 		Steps: []resource.TestStep{
 			resource.TestStep{
-				Config: testAccComputeTargetHttpsProxy_basic1,
+				Config: testAccComputeTargetHttpsProxy_basic(resourceName, 0, 0),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckComputeTargetHttpsProxyExists(
-						"google_compute_target_https_proxy.foobar"),
+						"google_compute_target_https_proxy.basic", &targetHttpsProxy),
 				),
 			},
 		},
 	})
 }
 
 func TestAccComputeTargetHttpsProxy_update(t *testing.T) {
+	var targetHttpsProxy compute.TargetHttpsProxy
+	var resourceName = fmt.Sprintf("httpsproxy-test-%s", acctest.RandString(10))
 
 	resource.Test(t, resource.TestCase{
 		PreCheck:     func() { testAccPreCheck(t) },
 		Providers:    testAccProviders,
 		CheckDestroy: testAccCheckComputeTargetHttpsProxyDestroy,
 		Steps: []resource.TestStep{
 			resource.TestStep{
-				Config: testAccComputeTargetHttpsProxy_basic1,
+				Config: testAccComputeTargetHttpsProxy_basic(resourceName, 0, 0),
```

Comment:
I totally rewrote this `TestAccComputeTargetHttpsProxy_update` since the old one was insufficient: it used to create a set of resources, then destroy them and recreate a new set of resources -- this is not a test of update!

Sorry for the ugly diff.